### DISCUSSION TITLE
Data Flow and the App object

### DISCUSSION THREAD

**Discussion by ghost** [Textualize/textual#2625] [Q&A] [answered]
On https://textual.textualize.io/guide/widgets/#data-flow it is written that Widgets should update their parents only by sending messages to them.
I wonder if this also applies to the App object itself.
Is it fine to call methods on the app object from it child Screens using self.app.* , or can this cause issues?

**Comment by davep** [comment, marked as answer]
If your widgets can be confident that `self.app` is of a type that has the methods you want to call, then you won't run into any problems. This does, however, mean your widgets won't be that portable; this may or may not be a concern.

My approach these days is to use the same data flow rules for anything involving the application class as with any other container. That's to say, ideally, I never have a widget call a method on `self.app` (other than those that are documented to be part of `App`) and instead send up messages.

**Comment by ghost** [comment]
Hi Dave,

thanks for your answer!
Since I'll only be using a single App it should be fine to call it directly then.

Maybe it would make sense to include the information you posted in the documentation?

Thank you very much!
